### PR TITLE
Feature/hierarchy builder

### PR DESCRIPTION
**Pull request by CarlHembrough**
### What

* Updated instance creation to also create the model for import tasks / hierarchy build tasks.
* Updated config to log as JSON with sensitive fields omitted.

### How to review

Review code

### Who can review

Anyone

### CODE REVIEW

File: models/models_test.go [hunk 1]
```diff
@@ -70,3 +70,37 @@ func TestCreateUploadedFileWithValidJson(t *testing.T) {
...
+			},
+		}
+
+		Convey("When CreateInstance is called", func() {
+
+			instance := CreateInstance(job, datasetID, datasetURL, codelists)
+
+			Convey("Then there should be a single build hierarchy task for the codelist that is marked as a hierarchy", func() {
+
+				So(instance.ImportTasks.BuildHierarchyTasks[0].State, ShouldEqual, CreatedState)
```
Comment (gedge) [inline models/models_test.go]: please add a check that there's only one task:
`So(len(instance.ImportTasks.BuildHierarchyTasks), ShouldEqual, 1)`

File: dataset/api.go [hunk 2]
```diff
@@ -33,10 +33,12 @@ func (api *API) CreateInstance(ctx context.Context, job *models.Job, recipeInst
 		return
 	}
 
-	logData["jsonUpload"] = jsonUpload
+	logData["jsonRequest"] = string(jsonUpload)
```
[the remark this answers is not in the document]
Comment (CarlHembrough) [reply dataset/api.go]: Changed all logData 'keys' to use snake case.

File: dataset/api.go [hunk 3]
```diff
@@ -33,10 +33,12 @@ func (api *API) CreateInstance(ctx context.Context, job *models.Job, recipeInst
...
 	jsonResult, httpCode, err := api.post(ctx, path, jsonUpload)
 	logData["httpCode"] = httpCode
-	logData["jsonResult"] = jsonResult
+	logData["jsonResponse"] = string(jsonResult)
```
[the remark this answers is not in the document]
Comment (CarlHembrough) [reply dataset/api.go]: Changed all logData 'keys' to use snake case.

File: models/models_test.go (hunk 1 above)
Comment (CarlHembrough) [reply models/models_test.go]: Added 👍

**Pull request merged**